include/boost/numeric/ublas/tensor/fixed_rank_extents.hpp

```diff
@@ -0,0 +1,238 @@
+//
+//  Copyright (c) 2018-2020, Cem Bassoy, cem.bassoy@gmail.com
+//  Copyright (c) 2019-2020, Amit Singh, amitsingh19975@gmail.com
+//
+//  Distributed under the Boost Software License, Version 1.0. (See
+//  accompanying file LICENSE_1_0.txt or copy at
+//  http://www.boost.org/LICENSE_1_0.txt)
+//
+//  The authors gratefully acknowledge the support of
+//  Google and Fraunhofer IOSB, Ettlingen, Germany
+//
+
+
+#ifndef _BOOST_NUMERIC_UBLAS_TENSOR_FIXED_RANK_EXTENTS_HPP_
+#define _BOOST_NUMERIC_UBLAS_TENSOR_FIXED_RANK_EXTENTS_HPP_
+
+#include <algorithm>
+#include <string>
+#include <initializer_list>
+#include <limits>
+#include <numeric>
+#include <stdexcept>
+#include <vector>
+#include <array>
```

Comment:
Can you please check if we need that many headers?

Reply:
the only redundant headers `string` and `numeric`

Reply:
why does `fixed_rank_extents` require `std::vector` ?

Reply:
it was to convert `fixed_rank_extents` to `dynamic_extents`.

Reply:
I recommend not to put a conversion operation inside `fixed_rank_extents` and define the free function where you need it. I remember that you needed it for some stride computations. I would put it in there as a free function or even just use `std::copy` if it is needed at that place.

Reply:
I removed it because I forgot I added a constructor that could accept any extents.